crates/metadata-store/src/grpc/client.rs

```diff
@@ -278,22 +278,22 @@ impl MetadataStore for GrpcMetadataStoreClient {
 fn map_status_to_read_error(status: Status) -> ReadError {
     match &status.code() {
         Code::Unavailable => ReadError::Network(status.into()),
-        _ => ReadError::Internal(status.to_string()),
+        _ => ReadError::RemoteError(RemoteError { status }),
```

Comment:
What is the `Internal` variant used for now? What if the remote implementation returns an `internal` error?